cmd/groups.go

```diff
@@ -33,26 +33,68 @@ func parseGroup(flags pflag.FlagSet) api.Group {
 var addGroupCmd = &cobra.Command{
 	Use:     "group",
 	Aliases: []string{"g"},
-	Short:   "Add a group to lagoon",
-	Run: func(cmd *cobra.Command, args []string) {
-		groupFlags := parseGroup(*cmd.Flags())
-		if groupFlags.Name == "" {
-			fmt.Println("Missing arguments: Group name is not defined")
-			cmd.Help()
-			os.Exit(1)
-		}
-		var customReqResult []byte
-		var err error
-		customReqResult, err = uClient.AddGroup(groupFlags)
-		handleError(err)
-		returnResultData := map[string]interface{}{}
-		err = json.Unmarshal([]byte(customReqResult), &returnResultData)
+	Short:   "Add a group to Lagoon, or add a group to an organization",
+	Long:    "To add a group to an organization, you'll need to include the `organization` flag and provide the name of the organization. You need to be an owner of this organization to do this.\nIf you're the organization owner and want to grant yourself ownership to this group to be able to deploy projects that may be added to it, specify the `owner` flag, otherwise you will still be able to add and remove users without being an owner",
+	PreRunE: func(_ *cobra.Command, _ []string) error {
+		return validateTokenE(lagoonCLIConfig.Current)
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		debug, err := cmd.Flags().GetBool("debug")
 		handleError(err)
+		groupName, err := cmd.Flags().GetString("name")
+		if err != nil {
+			return err
+		}
+		if err := requiredInputCheck("Group name", groupName); err != nil {
+			return err
+		}
+		orgOwner, err := cmd.Flags().GetBool("owner")
+		if err != nil {
+			return err
+		}
+		organizationName, err := cmd.Flags().GetString("organization")
+		if err != nil {
+			return err
+		}
+
+		current := lagoonCLIConfig.Current
+		token := lagoonCLIConfig.Lagoons[current].Token
+		lc := lclient.New(
+			lagoonCLIConfig.Lagoons[current].GraphQL,
+			lagoonCLIVersion,
+			lagoonCLIConfig.Lagoons[current].Version,
+			&token,
+			debug)
+
+		if organizationName != "" {
+			organization, err := l.GetOrganizationByName(context.TODO(), organizationName, lc)
+			handleError(err)
```

Comment:
In the commands that are `RunE` based, we should `return err` so these should be changed to return the err
```
if err != nil {
	return err
}
```

Only commands that are `Run` based should use `handleError` because they are old commands

Reply:
Ah yep I've updated all of these in https://github.com/uselagoon/lagoon-cli/pull/321, I'll update the relevant organization commands for this pr.